### DISCUSSION TITLE
How to Implement MaxPooling2D with Ceil Mode in Flax?

### DISCUSSION THREAD

**Discussion by firqaaa** [google/flax#2070] [Q&A] [answered]
What is the equivalent code for maxpooling2d with ceil mode in PyTorch?

So for example in PyTorch we can code like this : 
• nn.MaxPool2d(3, stride=2, padding=0, ceil_mode=True)
But in Flax, the framework only supports four parameters such as input, window_size, strides, and padding.

**Comment by andsteing** [comment]
Note that Flax also exports the more powerful [`nn.pool()`](https://flax.readthedocs.io/en/latest/_modules/flax/linen/pooling.html#pool) function; this can be used for example to implement `torch.nn.AvgPool2d` as described here:
https://github.com/google/flax/pull/1916

**Comment by marcvanzee** [comment, marked as answer]
Hi @firqaaa, we currently don't support ceil mode for our pooling layers. The best way to reproduce pytorch behavior is to use `VALID` padding and then pad the input manually to the shapes you want using the `padding` argument.
